### ISSUE TITLE
[no-unsafe-member-access & no-unsafe-call] false positives on values from getters

### ISSUE THREAD

**Issue by dzek69** [typescript-eslint/typescript-eslint#3294]
I'm not sure if this is a plugin or a parser problem. It does not crash but it seems more related to the parser (not giving the proper type to the rule) than actual rule. Sorry if I'm wrong.

- [x] I have tried restarting my IDE and the issue persists.
- [x] I have updated to the latest version of the packages.
- [x] I have [read the FAQ](https://github.com/typescript-eslint/typescript-eslint/blob/master/docs/getting-started/linting/FAQ.md) and my problem is not listed.

**Repro**

Demo repo:
https://github.com/dzek69/bug-typescript-eslint-unsafe-any-getter

- install deps (use yarn for version locks),
- run `yarn lint`

```JSON
{
  "rules": {
    "@typescript-eslint/no-unsafe-call": "error",
    "@typescript-eslint/no-unsafe-member-access": "error"
  },
  "parserOptions": {
    "sourceType": "module",
    "project": "./tsconfig.lint.json",
    "ecmaVersion": 9
  }
}
```

```TS
class Item {
    get data() {
        return {
            count: 6,
        }
    }
}

class Test {
    public get stuff() {
        return {
            list: [
                new Item(),
                new Item(),
                new Item(),
            ]
        }
    }

    public get numberStuff(): { list: Item[] } { // manually specifying the type is not helping
        return {
            list: [
                new Item(),
                new Item(),
                new Item(),
            ]
        }
    }
}

const x = new Test();

x.stuff.list.find(a => a.data.count < 3); // problem happens here
```

tsconfig: https://github.com/dzek69/bug-typescript-eslint-unsafe-any-getter/blob/master/tsconfig.lint.json

**Expected Result**

No errors should be given. Eslint should be able to get the types information from getters code

**Actual Result**

Two errors are happening here, both related to accessing data via getters, IDE and TypeScript don't have problems getting correct types here, Eslint does

```
  33:1   error  Unsafe call of an `any` typed value           @typescript-eslint/no-unsafe-call
  33:24  error  Unsafe member access .data on an `any` value  @typescript-eslint/no-unsafe-member-access
```

**Additional Info**

**Versions**

| package                     | version |
| --------------------------- | ------- |
| `@typescript-eslint/parser` | `4.22.0` |
| `TypeScript`                | `4.2.4` |
| `ESLint`                    | `7.24.0` |
| `node`                      | `15.2.0` |

**Comment by bradzacher** [comment]
https://github.com/dzek69/bug-typescript-eslint-unsafe-any-getter/blob/master/tsconfig.lint.json#L9

With absolutely no `lib`s defined - you will not even have the basic JS types.
Which means that TS doesn't have a reference to the `Array` type.

So your return type resolves to `{ list: any }` where that `any` is an "error any".

You can see this in your IDE by doing the following to your example repo:
1) rename `tsconfig.lint.json` to `tsconfig.json`
2) comment out the `isolatedModules` config
3) restart IDE or the TS server

You'll see an error in the tsconfig file:
![image](https://user-images.githubusercontent.com/7462525/115275609-df149880-a0f6-11eb-81ab-5b2bc2587b6b.png)
